tests/realm.cpp

```diff
@@ -1136,3 +1142,105 @@ TEST_CASE("SharedRealm: compact on launch") {
     results.async([](std::exception_ptr) { });
 }
 #endif
+
+TEST_CASE("SharedRealm: update_schema with initialization_function")
+{
+    TestFile config;
+    config.schema_version = 0;
+    config.schema = Schema{
+        {"object", {{"value", PropertyType::String, "", "", false, false, false}}},
+    };
+    bool initialization_function_called = false;
+    Realm::DataInitializationFunction initialization_function = [&initialization_function_called](auto shared_realm) {
+        REQUIRE(shared_realm->is_in_transaction());
+        initialization_function_called = true;
+    };
+
+    SECTION("Initialization function should be called for unversioned realm") {
+        config.initialization_function = initialization_function;
+        SECTION("Automatic mode")
```

Comment:
Structuring the tests like this with the actual checks being done after the leaf section results in test failures not actually reporting which section failed.